### ISSUE TITLE
Vosk Speech to test WOH cannot get language code from Dubin core metadata

### ISSUE THREAD

**Issue by fishfree** [opencast/opencast#3712]
I insert the vosk speech-to-text operation in the partial-publish.xml in my 11.5 Opencast as below:
```
<!-- Create transcript with vosk-cli -->
    <operation
        id="speechtotext"
        description="Generates subtitles for video and audio files">
       <configurations>
          <configuration key="source-flavor">*/source</configuration>
          <configuration key="target-flavor">captions/vtt+auto</configuration>
          <configuration key="target-element">attachment</configuration>
          <configuration key="target-tags">archive,subtitle,engage-download</configuration>
          <configuration key="language-code">${catalogs['episode']['language']}</configuration>
        </configurations>
    </operation>

<!-- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -->
    <!-- Publish to publication channels                                   -->
    <!--                                                                   -->
    <!-- Send the encoded material along with the metadata to the          -->
```

However, it still use the fallback eng language-code. Why it cannot read the language code from the Dublin core metadata? I read there is [such related code here](https://github.com/opencast/opencast/blob/61f1c644ca89bc470e5757fd5ff6911df98c6e68/modules/speech-to-text-workflowoperation/src/main/java/org/opencastproject/workflow/handler/speechtotext/SpeechToTextWorkflowOperationHandler.java#L265). But I'm still very confused by many concepts in Opencast.

**Comment by karendolan** [comment]
HI @fishfree 
[BIG EDIT] to avoid confusing other readers, I'm editing the response to correct it. 

Our site uses that style to pull values from the episode catalog, but we include a fallback value, example `<configuration key="subject">${catalogs['episode']['type']!'type missing'</configuration>`. 
But, we only use in the send email WOH to support FreeMarker templates.
 https://github.com/opencast/opencast/blob/develop/modules/notification-workflowoperation/src/main/java/org/opencastproject/workflow/handler/notification/EmailWorkflowOperationHandler.java#L211-L224

**Comment by karendolan** [comment]
Document reference for the FreeMarker uses
https://github.com/opencast/opencast/blob/develop/docs/guides/admin/docs/workflowoperationhandlers/publish-configure-woh.md
https://github.com/opencast/opencast/blob/729fecb21a932466e7ed70af1c6de68b8bd9cde9/docs/guides/admin/docs/workflowoperationhandlers/send-email-woh.md

**Comment by fishfree** [comment]
So, It would be much better to use the catalog data in any WOH.

**Comment by Arnei** [comment]
This issue should have been resolved by #3806. If not, please reopen the issue.
